README.md

```diff
@@ -0,0 +1,224 @@
+below.
+
+```bash
+ninja
+python3 -m pytest test/  # (Optional, but recommended) run the tests
+sudo ninja install
+```
+
+### 2. Detailed instructions
+
+_Tested 22 Dec. 2020 on Ubuntu 20.04._
+
+1. Download and `cd` into the source code
+    1. To download the latest source code:
+        ```bash
+        git clone https://github.com/libfuse/sshfs.git
+        cd sshfs
+        ```
+    1. OR (recommended): To download the latest `sshfs` _release_:
+        1. Find the release you want here: https://github.com/libfuse/sshfs/releases.
+        1. Find the link to the `*.tar.gz` file you want. Example: https://github.com/libfuse/sshfs/archive/sshfs-3.7.1.tar.gz.
+        1. Download, extract, and `cd` into it:
+            ```bash
+            wget https://github.com/libfuse/sshfs/archive/sshfs-3.7.1.tar.gz
+            tar -xvzf sshfs-3.7.1.tar.gz
+            cd sshfs-sshfs-3.7.1
+            ```
+1. Install dependencies
+    ```bash
+    sudo apt update
+    sudo apt install meson cmake fuse3 libfuse3-dev libglib2.0-dev
+    ```
+1. Create a **build** dir, `cd` into it, run `meson`, and build and link with `ninja`:
+    ```bash
+    mkdir build 
+    cd build 
+    meson ..
+    ninja  # this builds and produces the new `sshfs` executable
+    ```
+<!-- Important: this HTML anchor is referenced inside "test_sshfs.py". If you change its name here,
+change it there too. -->
+<a id="sshkeygen"></a> 
+1. (Optional, but recommended) test your new `sshfs` executable. [This also includes detailed notes
+   about ssh key generation and setup].
+    1. Note: ssh key generation notes come [from GitHub here](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).
+    ```bash
+    # install the `pytest` python3 module in case you don't already have it
+    pip3 install pytest  
+    # install `sshd`
+    sudo apt update 
+    sudo apt install openssh-server
+
+    # Configure passwordless ssh key-based login so you can ssh into yourself
+    # (via `localhost`) for testing; see GitHub link above.
+    # 1. Generate a new public-private key pair. 
+    ssh-keygen -t ed25519 -C "your_email@example.com"
+    eval "$(ssh-agent -s)"
+    ssh-add ~/.ssh/id_ed25519
+    # 2. Now add your new "~/.ssh/id_ed25519.pub" public key to your 
+    # "~/.ssh/authorized_keys" file so you can have key-based password-less
+    # ssh sessions into yourself via `localhost` for testing.
+    cat ~/.ssh/id_ed25519.pub >> ~/.ssh/authorized_keys 
+
+    # run the python3 tests in the "test" dir
+    python3 -m pytest test/
+    ```
+    For [sample test output, see here](test/README.md).
+1. Install `sshfs`:
+    1. Normal method:
+        ```bash
+        sudo ninja install
+        ```
+    1. "Light" method. This technique simply creates a symlink to the executable in your `~/bin` 
+       dir so your Linux distro's install still remain's intact and untouched. (For Ubuntu).
```

Comment:
This is the case also when using ninja install. Your system's sshfs is in /usr/bin, ninja install into /usr/local/bin.

Reply:
What's the ninja uninstall command?

Reply:
I don't know. But I also don't see the connection to what I'm saying...?

Reply:
> If you install in /usr/locl, then your Linux distro's install remains intact and untouched.

Agreed. So, if someone wants to go back to using their Linux distro's version of `sshfs`, how do they do that? The logical answer is: run the ninja uninstall command. That's why I asked "what's the ninja uninstall command?" I'd like to write it down here too. 

For the "Alternative method" here, I show: `rm ~/bin/sshfs`.

A few options to uninstall the `ninja install` might be:
```bash
rm /usr/local/bin/sshfs
```
Or: modify the path to include `/urs/local/bin` *after* `/usr/bin` instead of before it (probably not recommended), or (best, if such a thing exists): `sudo ninja uninstall`.

Reply:
Have you tried running `ninja uninstall`? If I understand https://mesonbuild.com/Release-notes-for-0-38-0.html correctly, then this should exist and is probably the best approximation to a true uninstall that we have.